test/test_encoding.py

```diff
@@ -0,0 +1,743 @@
+#!/usr/bin/python3 -u
+"""
+SPDX-License-Identifier: BSD-2
+"""
+import unittest
+
+from tpm2_pytss import *
+from tpm2_pytss.encoding import (
+    base_encdec,
```

Comment:
So what's base_encdec all hex strings? If so perhaps call this hex_encdec

Reply:
it's the base class for other encoders/decoders, so it's the minimal viable implementation so subclasses can just implement the needed changes

Reply:
Is it supposed to be public? If it's public the name should be more descriptive of what it does IMO.

Reply:
I would prefer it to be public, if users want to implement their own encoders/decoders
I could always rename it to minimal_encdec and improve the description in the docstring